src/Craftify.Geometry.Domain/LineSegmentGraphExtensions.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace Craftify.Geometry.Domain;
+
+public static class LineSegmentGraphExtensions
+{
+    public static IReadOnlyList<Point3D> DistinctedNodes(this LineSegmentsGraph graph)
```

Comment:
In English the word Distinct is an adjective, so I think the correct way of naming the method would be `DistinctNodes`